File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/base/Motion.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.base;

/**
 * @author Norihiro
 *
 */
public class Motion {
	public String hashName; 
	public boolean loaded;
	public Object onload;
	public Motion[] motionArray;
	public int frameLength;
	public int defaultFPS;
	public boolean loop;
	
	public Motion(){
		
	}
	public Motion clone(){
		return this;
	}
	public void copy(Motion motion){
		
	}
	
}

File: src/js/base/Motion.js
'use strict'

/**
 * Motion class
 * @access public
 */
export default class Motion {
  /**
   * constructor
   * @access public
   * @constructor
   */
  constructor() {
    this.hashName = ''
    this.loaded = false
    this.onload = null

    // motion
    this.motionArray = new Map()

    this.frameLength = 0
    this.defaultFPS = 0
    this.loop = false
  }

  clone() {
    return this
  }

  destroy() {
  }

  copy(motion) {
    this.motionArray = motion.motionArray
    this.frameLength = motion.frameLength
    this.defaultFPS  = motion.defaultFPS
    this.loop        = motion.loop
  }
}
